src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -623,41 +623,53 @@ namespace shogun
 		}
 
 		/**
-		 * Compute the eigenvalues and eigenvectors of a symmetric matrix.
+		 * Compute the top-k eigenvalues and eigenvectors of a symmetric matrix.
```

Comment:
Mmmmmh. I don't think that is a good API to offer.
If you compute all of them anyways, you can just return all, and then the caller picks the ones he wants.
There is also iterative methods that are much faster when only computing a few, and those are the ones that should get this API here, with a warning label ("slower for computing all")